### PR TITLE
[Jira] Documentation for Jira connector

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## This PR addresses the following changes:

- Added Jira connector in the [developing.md](https://github.com/elastic/connectors-python/blob/main/docs/DEVELOPING.md).

#### Pre-Review Checklist
- [x] Considered corresponding documentation changes

### CODE REVIEW

File: docs/DEVELOPING.md
Comment (timgrein) [inline docs/DEVELOPING.md:449]: Maybe we could provide some more info, why this is useful? Speed-up etc.

File: docs/DEVELOPING.md
Comment (danajuratoni) [inline docs/DEVELOPING.md:373]: Updated the TDD to reflect the latest update, that Project Keys should be a configurable parameter. AFAIK we can update the doc right away and add it in an enhancement.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:369]: Username is just for Jira server. Need to mention `service account` for Jira cloud.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:370]: Password is used for Jira server. Need to mention `API key` for Jira cloud.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:383]: Here, `username` is just used for Jira server. User will configure `service account` for Jira cloud, so I've already mentioned the same in "Gather Jira details".

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:387]: `Password` is used for Jira server only. `API key` is responsible for authenticating to Jira cloud.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:373]: Sure, I will update the TDD and doc with accommodating a new configuration parameter `projects`.

Comment (danajuratoni) [reply docs/DEVELOPING.md:374]: Thank you!

File: docs/DEVELOPING.md
Comment (danajuratoni) [inline docs/DEVELOPING.md:405]: Please make this parameter header inline with the others (have the same level)

**Comment by danajuratoni** [comment]
Looks good to me, with a few mentions:
- please mention the supported versions for Jira Server 
- default values will be reworked as part of the Rich Configurable Fields initiative (as well as other field attributes)
- am I right to assume the documentation for configuring Jira Server is still applicable? https://www.elastic.co/guide/en/workplace-search/current/workplace-search-jira-server-connector.html#jira-server-configuration

cc: @leemthompo @chriscressman  @ppf2

**Comment by moxarth-elastic** [comment]
> am I right to assume the documentation for configuring Jira Server is still applicable? 

There's no need to configure the Jira server as per the doc. Just require username and password to setup the connector for Jira server.

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:833]:
```suggestion
Whether the connector should extract content from a Jira attachment. Default value is `True` i.e. the connector will try to extract file contents.
```

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by moxarth-elastic** [comment]
@timgrein @leemthompo Can you please provide your feedback since the functionality PR has been merged? cc: @danajuratoni

**Comment by danajuratoni** [comment]
LGTM
cc: @seanstory

File: docs/DEVELOPING.md
Comment (seanstory) [inline docs/DEVELOPING.md:829]: I believe this was just bumped to 100 in another PR.

File: docs/DEVELOPING.md
Comment (seanstory) [inline docs/DEVELOPING.md:837]: I believe these sections were recently removed from all docs, because this is controlled through a separate mechanism

**Review by seanstory — COMMENTED** [review]
Two small nits. Otherwise, looks good to me.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:829]: Yes, let me update it.

**Comment by moxarth-elastic** [comment]
@danajuratoni I've updated the documentation structure as per the released connectors (HTML format). You can have a look at it.

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:58]:
```suggestion
Dropdown to determine Jira platform type: `Jira Cloud` or `Jira Server`. Default value is `Jira Cloud`.
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:83]:
```suggestion
##### `projects_keys`
```

Double check this

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:85]: We have `projects`, `Project Keys`, and `project keys` here. Which one is correct?

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:35]: We can just call this Section `### Configuration` per the released docs format and remove the sub-steps 1 and 2.

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:40]:
```suggestion
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:50]:
```suggestion
```

Comment (leemthompo) [reply docs/DEVELOPING.md:50]: We can remove this per the released docs format. All this information is contained in the configuration fields section. However, if any future docs have details here that pertain to the third-party data source, this should go in the **Usage** section.

**Review by leemthompo — COMMENTED** [review]
Looks good, just a few minor edits to be made before we can merge this.

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:21]:
```suggestion
This connector is available as a **connector client** from the **Python connectors framework**. This connector is in **beta** and is subject to change. To use this connector, satisfy all [connector client requirements](https://www.elastic.co/guide/en/enterprise-search/master/build-connector.html).
```

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:33]: It will be more clear if we rephrase the compatibility section to be something like:

Jira Cloud or Jira Server/Data Center versions 7 or later are compatible with Elastic connector frameworks.

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:58]: Do we support Jira Data Center in addition to Jira Server?  If so, let's change all our "Jira Server" references to "Jira Server/Data Center".

File: docs/DEVELOPING.md
Comment (ppf2) [reply docs/DEVELOPING.md:85]: Providing a link to what project key means will be helpful: https://support.atlassian.com/jira-software-cloud/docs/what-is-an-issue/#Workingwithissues-Projectkeys

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:83]: Can we specify what type of objects/entities are supported by this connector? Does it fetch _everything_ within a project including issues, comments, attachments, linked issues, components, fix/affected versions, watchers, etc..?

In general, I think every connector documentation should have a dedicated section that lists the objects/entities we support (something to add to our "template") 😄  cc: @leemthompo @danajuratoni

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:85]: @leemthompo `projects` is just a configuration parameter used in the connector code. User will see a label i.e. `Jira Project Keys` and will configure `project keys`. However, i'll provide a link for `project key` as @ppf2 suggested.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md:83]: Since, We've finalized an approach to index everything we get from the Atlassian API Response. Yes, this connector supports all entities that API is providing.

File: docs/DEVELOPING.md
Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: No, the connector supports Jira Cloud and Jira Server only.

File: docs/DEVELOPING.md
Comment (leemthompo) [reply docs/DEVELOPING.md:68]: > I think every connector documentation should have a dedicated section that lists the objects/entities we support (something to add to our "template") 

This information should go in a **Documents and syncs** subheading which describes what gets synced and, if helpful, what _doesn't_ get synced. 😊

Comment (moxarth-elastic) [reply docs/DEVELOPING.md:68]: @leemthompo I've added a section `Documents and Sync` to include this information.

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:105]:
```suggestion
The connector syncs the following objects and entities:
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:106]:
```suggestion
- **Projects**
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:108]:
```suggestion
- **Issues**
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:111]:
```suggestion
- **Attachments**
```

File: docs/DEVELOPING.md
Comment (leemthompo) [inline docs/DEVELOPING.md:113]:
```suggestion
**Note:** Archived projects and issues are not indexed.
```

File: docs/DEVELOPING.md
Comment (leemthompo) [reply docs/DEVELOPING.md]: OK but let's not confuse the reader by enclosing everything in backticks here.  :)

**Comment by leemthompo** [comment]
@moxarth-elastic just a couple of final edits, let me know when you've completed them and I'll get this approved. :)

**Review by leemthompo — APPROVED** [review]
LGTM

**Comment by moxarth-elastic** [comment]
@ppf2 I've addressed your comments, can you please take a look?

File: docs/DEVELOPING.md
Comment (ppf2) [inline docs/DEVELOPING.md:33]: It might also be helpful to explicitly call out that Data Center editions are not currently supported.  @danajuratoni WDYT?

Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: @ppf2 Yes, Thanks. I've called it out explicitly.

Comment (moxarth-elastic) [reply docs/DEVELOPING.md]: > @ppf2 Yes, Thanks. I've called it out explicitly.

@ppf2 Did you get a chance to look at this change?

**Comment by moxarth-elastic** [comment]
@ppf2 I have addressed your comments. The PR has been approved by other reviewers and is waiting for your approval. Can you please go through it?

**Review by ppf2 — APPROVED** [review]
Thx!

**Review by wangch079 — APPROVED** [review]

**Comment by wangch079** [comment]
buildkite test this

**Pull request merged**
